src/librustc_typeck/check/mod.rs

```diff
@@ -1005,6 +1042,8 @@ fn typeck_tables_of_with_fallback<'tcx>(
     let body = tcx.hir().body(body_id);
 
     let tables = Inherited::build(tcx, def_id).enter(|inh| {
+        let in_const = inh.in_const_context.get();
+        inh.in_const_context.set(in_const_context);
```

Comment:
instead of passing this in as a flag, can you try using `tcx.describe_def(def_id)` to figure out whether we are in a const context? or maybe even not have the flag at all and check that at the diagnostic emitter site?

Reply:
There's no `describe_def`?

Yeah, that seems like a reasonable approach, I just need to verify it is doable (we go through enough layers of indirection that expect success that it might be tricky).

Reply:
I used `is_const_context(HirId)` which after making a small fix is appropriate for this check and removed the sentinel.